internal/convert/clouddriver_test.go

```diff
@@ -26,38 +26,50 @@ import (
 	"google.golang.org/protobuf/proto"
 )
 
-var halToClouddriverTests = []struct {
-	name string
-	hal  *config.Hal
-	want *config.Clouddriver
-}{
-	{
-		"Empty hal config",
-		&config.Hal{},
-		&config.Clouddriver{},
-	},
-	{
-		"Empty providers",
-		&config.Hal{
-			Providers: &cloudprovider.Providers{},
+var clouddriverTests = configTest{
+	generator: func(h *config.Hal) proto.Message { return convert.HalToClouddriver(h) },
```

Comment:
This is a place where I miss generics.  What my Java brain really wants is for `generator` to be of type:
```
<T extends proto.Message> func(h *config.Hal) T
```
If that were the type, then I could just pass `convert.HalToClouddriver` directly; but since I can't write that I need to write a new function of signature `func(h *config.Hal) proto.Message` that just calls `convert.HalToClouddriver` (and relies on implicitly converting to the interface in the return).